qxf2/qxf2-survey
# Add authentication

This PR adds an authentication form for the admin page
 
Prerequisite:
1.  I will email a  few CSV files, you will have to store those files in a folder anywhere on your computer
2.  I will also mail the Admin credentials that have to be stored in the `AdminCredentials.js` file

---
Steps to setup:
---
1. Clone the repository
2. Switch to the `Add_Authentication` branch

---
Setting up backend
---
1. Download and install neo4j  desktop https://neo4j.com/download-v2/
2. Create a new project and add a local database. Set the database name as "neo4j" and the password as "TestQxf2"
3. After the database is created, click on the ... option on your database and select Manage
4. Go to Settings tab
5. Find dbms.directories.import and assign it the path of your CSV files mentioned in Prerequisite step 1. (ex: dbms.directories.import=C:/Users/xyz/Desktop/csv_files). Click Apply to save the changes.
6. Start the database
7. In bash, cd into the backend directory.
8. Setup a virtual environment and activate it
9. Install the requirements with pip install -r requirements.txt
10. In your backend directory rename .env.example to .env
11. Edit the .env and make the following changes:
    ```
    DATABASE_USERNAME = "neo4j"
    DATABASE_PASSWORD = "TestQxf2"
    DATABASE_HOST = "bolt://localhost:7687"
    API_KEY = "qxf2"
    PROJECT_NAME= "Help Survey Project"
    ```
12. cd into the `backend/app` directory
13.  Kick-Start the Application with the following command: `uvicorn main:app --reload`
14. Run `python backend/app/db/init_db.py` - which imports the csv files into the database  
----
 Setting up Frontend
---

1. cd into the frontend directory.
2. Install NodeJS from here: https://nodejs.org/en/
3. Ensure that NodeJS and NPM has been installed by running the following command: `node --version` and `npm --version`
4. If a specific version is popped up, run the command: `npm install` to install all dependencies.
5. Rename the `.env.example` file to `.env` 
6. Edit the `.env` file and make the following change
  ```
  REACT_APP_API_KEY = "qxf2"
  ````
7. Edit the file `frontend/src/AdminCredentials.js` add the credentials I have sent through email in Prerequisite step 2.
8. Edit the file `frontend/src/pages/data/urlConf.js` and change the `URL` to `"http://localhost:8000"` in order to connect and fetch the data from backend
9. `cd` back to `frontend` directory and run the command: npm start to kick-start the Application. It will take a couple of seconds to setup the development server on Port 3000.
